cognite/client/_api/postgres_gateway/users.py

```diff
@@ -0,0 +1,274 @@
+        )
+
+    def delete(self, username: str | SequenceNotStr[str], ignore_unknown_ids: bool = False) -> None:
+        """`Delete postgres user(s) <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/delete_users>`_
+
+        Delete postgres users
+
+        Args:
+            username (str | SequenceNotStr[str]): Username to authenticate the user on the DB.
+            ignore_unknown_ids (bool): Ignore usernames that are not found
+
+
+        Examples:
+
+            Delete user:
+
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> client.postgres_gateway.users.delete.delete(["myUser", "myUser2"])
+
+
+        """
+        self._warning.warn()
+        extra_body_fields: dict[str, Any] = {}
+        extra_body_fields["ignore_unknown_ids"] = ignore_unknown_ids
+
+        self._delete_multiple(
+            identifiers=UsernameSequence.load(usernames=username),
+            wrap_ids=True,
+            returns_items=False,
+            extra_body_fields=extra_body_fields or None,
+            headers={"cdf-version": "beta"},
+        )
+
+    @overload
+    def retrieve(self, username: str, ignore_unknown_ids: bool = False) -> User: ...
+
+    @overload
+    def retrieve(self, username: SequenceNotStr[str], ignore_unknown_ids: bool = False) -> UserList: ...
+
+    def retrieve(self, username: str | SequenceNotStr[str], ignore_unknown_ids: bool = False) -> User | UserList:
+        """`Retrieve a list of users by their usernames <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/retreive_users>`_
+
+        Retrieve a list of postgres users by their usernames, optionally ignoring unknown usernames
+
+        Args:
+            username (str | SequenceNotStr[str]): Username to authenticate the user on the DB.
+            ignore_unknown_ids (bool): Ignore usernames that are not found
+
+        Returns:
+            User | UserList: A user
+
+        Examples:
+
+            Retrieve user:
+
+                    >>> from cognite.client import CogniteClient
+                    >>> client = CogniteClient()
+                    >>> res = client.postgres_gateway.users.retrieve("myUser", ignore_unknown_ids=True)
+
+        """
+        self._warning.warn()
+
+        return self._retrieve_multiple(
+            list_cls=UserList,
+            resource_cls=User,
+            identifiers=UsernameSequence.load(usernames=username),
+            ignore_unknown_ids=ignore_unknown_ids,
+            headers={"cdf-version": "beta"},
+        )
+
+    def list(self, limit: int = DEFAULT_LIMIT_READ) -> UserList:
+        """`Fetch scoped users <https://api-docs.cognite.com/20230101-beta/tag/Postgres-Gateway-Users/operation/filter_users>`_
+
+        List all users in a given project. If more than `limit` users exist, a cursor for pagination will be returned
+        with the response.
```

Comment:
Pretty sure we will page through all the results automatically:
```suggestion
        List all users in a given project.
```